src/plugins/mpi/pmix/pmixp_client_v2.c

```diff
@@ -88,12 +88,37 @@ static pmix_status_t _abort_fn(const pmix_proc_t *proc, void *server_object,
 {
 	/* Just kill this stepid for now. Think what we can do for FT here? */
 	PMIXP_DEBUG("called: status = %d, msg = %s", status, msg);
-	slurm_kill_job_step(pmixp_info_jobid(), pmixp_info_stepid(), SIGKILL);
 
-	if (NULL != cbfunc) {
-		cbfunc(PMIX_SUCCESS, cbdata);
+	int abort_port = pmixp_info_abort_port();
```

Comment:
Implement as a separate function:
`pmixp_abort_send()`

Reply:
Improve client-side socket code (leverage Slurm protocol API):
* setup address from (char*) hostname & port [[link](https://github.com/BorisBochkaryov/slurm/blob/master/src/common/slurm_protocol_socket.c#L637)]
* Open stream [[link](https://github.com/BorisBochkaryov/slurm/blob/master/src/common/slurm_protocol_socket.c#L455)]
* Send the message with timeout [[link](https://github.com/BorisBochkaryov/slurm/blob/master/src/common/slurm_protocol_socket.c#L202)]